core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -0,0 +1,241 @@
+
+    val deniedPrincipals = getPrincipals(opts, opts.denyPrincipalsOpt)
+
+    val allowedHosts = getHosts(opts, opts.allowHostsOpt, opts.allowPrincipalsOpt)
+
+    val deniedHosts = getHosts(opts, opts.denyHostssOpt, opts.denyPrincipalsOpt)
+
+    val operations = if (opts.options.has(opts.operationsOpt))
+      opts.options.valueOf(opts.operationsOpt).toString.split(delimeter).map(s => Operation.fromString(s)).toSet
+    else
+      Set[Operation](All)
+
+    var acls = new collection.mutable.HashSet[Acl]
+    if (allowedHosts.nonEmpty && allowedPrincipals.nonEmpty)
+      acls ++= getAcls(allowedPrincipals, Allow, operations, allowedHosts)
+
+    if (deniedHosts.nonEmpty && deniedPrincipals.nonEmpty)
+      acls ++= getAcls(deniedPrincipals, Deny, operations, deniedHosts)
+
+    acls.toSet
+  }
+
+  def getAcls(principals: Set[KafkaPrincipal], permissionType: PermissionType, operations: Set[Operation],
+              hosts: Set[String]): Set[Acl] = {
+    var acls = new collection.mutable.HashSet[Acl]
+    for {
+      principal <- principals
+      operation <- operations
+      host <- hosts
+    } acls += new Acl(principal, permissionType, host, operation)
+
+    acls.toSet
+  }
+
+  private def getHosts(opts: AclCommandOptions, hostOptionSpec: ArgumentAcceptingOptionSpec[String],
+                       principalOptionSpec: ArgumentAcceptingOptionSpec[String]): Set[String] = {
+    if (opts.options.has(hostOptionSpec))
+      opts.options.valueOf(hostOptionSpec).toString.split(delimeter).toSet
+    else if (opts.options.has(principalOptionSpec))
+      Set[String](Acl.WildCardHost)
+    else
+      Set.empty[String]
+  }
+
+  private def getPrincipals(opts: AclCommandOptions, principalOptionSpec: ArgumentAcceptingOptionSpec[String]): Set[KafkaPrincipal] = {
+    if (opts.options.has(principalOptionSpec))
+      opts.options.valueOf(principalOptionSpec).toString.split(delimeter).map(s => KafkaPrincipal.fromString(s)).toSet
+    else
+      Set.empty[KafkaPrincipal]
+  }
+
+  private def getResource(opts: AclCommandOptions): Resource = {
+    val resource: Resource = if (opts.options.has(opts.topicOpt))
+      new Resource(Topic, opts.options.valueOf(opts.topicOpt).toString)
+    else if (opts.options.has(opts.clusterOpt))
+      Resource.ClusterResource
+    else if (opts.options.has(opts.groupOpt))
+      new Resource(ConsumerGroup, opts.options.valueOf(opts.groupOpt).toString)
+    else
+      null
+
+    if (resource == null)
+      CommandLineUtils.printUsageAndDie(opts.parser, "You must provide at least one resource: --topic <topic> or --cluster or --consumer-group <group>")
+
+    resource
+  }
+
+  private def confirmaAction(msg: String): Boolean = {
+    println(msg)
+    val userInput: String = Console.readLine()
+
+    "y".equalsIgnoreCase(userInput)
+  }
+
+  class AclCommandOptions(args: Array[String]) {
+    val parser = new OptionParser
+    val config = parser.accepts("config", "REQUIRED: Path to server.properties file.")
```

Comment:
Added --authorizer --authorizer-property. 

--authorizer-property does not follow messageFormatterArgOpt which seem to assume only one property will be specified as oppose to in case of authorizer where multiple properties can be specified.